src/artm/utility/progress_printer.cc.in

```diff
@@ -48,7 +42,7 @@ void ProgressPrinter::Set(size_t pos) noexcept {
   int pct = static_cast<int>((pos_ * 100) / max_);
   if (pct != prevpct) {
     // ANSI code \033[<N>D moves cursor to N positions backwards
-    fprintf(stderr, " %3d%%\033[5D", pct);
+    fprintf(stderr, " %3d%%\b\b\b\b\b", pct);
```

Comment:
Here is the problem. \b and \033[ND work different. The former erases symbols while the latter moves cursor without erasing anything. So now the percentage gets removed at once, effectively nothing is printed either on Windows or POSIX. That's why I didn't want \b: you are going to have an internal state in ProgressPrinter, so that the very first fprintf is without \b-s but all the subsequent are. You cannot encode that state into pos_ because there may be Set(0). Please note that placing \b in front of %3d%% obviously will not do the trick.
I would add the Windows 10 support instead. After all, 7 is slowly dying and 8's share is tiny.